### DISCUSSION TITLE
Forced Alignment

### DISCUSSION THREAD

**Discussion by qo4on** [NVIDIA-NeMo/NeMo#2657] [Q&A]
Hi!
I have an audio and a text of it. Can NeMo return timestamps for every word in the text file? I believe `AlignerModel` is what I need. What is the right way to use it?

```
import nemo.collections.tts as nemo_tts
tts_models = [model for model in dir(nemo_tts.models) if model.endswith("Model")]

['AlignerModel',
 'DegliModel',
 'EDMel2SpecModel',
 'FastPitchHifiGanE2EModel',
 'FastPitchModel',
 'FastSpeech2HifiGanE2EModel',
 'FastSpeech2Model',
 'GlowTTSModel',
 'GriffinLimModel',
 'HifiGanModel',
 'MelGanModel',
 'MelPsuedoInverseModel',
 'SqueezeWaveModel',
 'Tacotron2Model',
 'TalkNetDursModel',
 'TalkNetPitchModel',
 'TalkNetSpectModel',
 'TwoStagesModel',
 'UniGlowModel',
 'WaveGlowModel']
```

**Comment by qo4on** [comment]
Is there anyone alive?

**Reply by PatrykNeubauer** [reply]
If I'm not mistaken, the aligner model is designed for use with TTS models as an alignment learning framework. 
Not sure, if there's a simple way to get what you want out of NeMo, but [Montreal Forced Aligner](https://montreal-forced-aligner.readthedocs.io/en/latest/) can do exactly what you want easily. 
It's even used by NeMo in it's scripts for ljspeech preprocessing

**Reply by titu1994** [reply]
Yup, we're alive. MFA is very well supported for exactly this task. We're working on aligners, but that will take time.
